lib/puppet/type/rabbitmq_binding.rb

```diff
@@ -11,21 +11,48 @@
     end
   end
 
+  # Match patterns without '@' as arbitrary names; match patterns with
+  # src@dst@vhost to their named params for backwards compatibility.
+  def self.title_patterns
+      [
+        [ /^(\S+)@(\S+)@(\S+)$/m, 
+          [ [:source], [:dest], [:vhost] ]
+        ],
+        [ /(^([^@]*)$)/m, [ [:name] ] ]
+      ]
+  end
+
   newparam(:name, :namevar => true) do
     desc 'source and destination of bind'
-    newvalues(/^\S*@\S+@\S+$/)
+  end
+
+  newparam(:source, :namevar => true) do
+    desc 'source of binding'
+    newvalues(/^\S+$/)
+  end
+
+  newparam(:dest, :namevar => true) do
+    desc 'destination of binding'
+    newvalues(/^\S+$/)
+  end
+
+  newparam(:vhost, :namevar => true) do
+    desc 'vhost'
+    newvalues(/^\S+$/)
+    defaultto('/')
```

Comment:
Comment: keep the default because when explicitly specifying source/dest/routing_key and not using title_pattern, then it's good to have a default.